### ISSUE TITLE
Set the fixity of _<?_ in Data.Nat.Properties

### ISSUE THREAD

**Issue by MatthewDaggitt** [agda/agda-stdlib#207]
Set the fixity of _<?_ in Data.Nat.Properties.

**Comment by MatthewDaggitt** [comment]
Closed with 5f4c67eca643120a275bea894716c76cc9f2defc

**Comment by gallais** [comment]
Using the new `-Wall` option which I have just introduced, I have
found a lot more mixfix definitions which do not come with a fixity
declaration:

https://gist.github.com/gallais/79386a6ba8826620dc82b3f5d5af2bde

**Comment by andreasabel** [comment]
Well, bracketing constructs like `[_]` or `[_,_]` do not need a fixity, in fact, since they are neither pre/post nor infix operators, a precedence or associativity value for these does not make sense.
The warning mechanism should be adapted to disregard these.

**Comment by gallais** [comment]
I've updated the gist with the new set of warnings ignoring well-bracketed mixfix decls

**Comment by MatthewDaggitt** [comment]
As of Agda 2.6.0 the warnings for the missing fixities should now work properly, so this should be much more doable.

**Comment by MatthewDaggitt** [comment]
Particularly egregious example in `Relation.Binary.Construct.NonStrictToStrict`

**Comment by JacquesCarette** [comment]
@gallais can you update the gist please? I think I'm going to spend some of my time working on this.

**Comment by gallais** [comment]
@JacquesCarette It's just a matter of running `agda` on Everything with `-Wall`.

**Comment by Saransh-cpp** [comment]
Hi, (I don't know if I am doing it right) I ran -
```
agda -Wall Everything.agda
```
locally, but could not reproduce the list of warnings, instead I got an error saying -
```
/Users/saransh/Code/Mitacs/agda-stdlib/Everything.agda:11,8-18
The name of the top level module does not match the file name. The
module Everything should be defined in one of the following files:
  /Users/saransh/Code/Mitacs/agda-stdlib/src/Everything.agda
  /Users/saransh/Code/Mitacs/agda-stdlib/src/Everything.lagda
  /opt/homebrew/Cellar/agda/2.6.3/libexec/ghc-9.4.4/Agd-2.6.3-d4728884/share/lib/prim/Everything.agda
  /opt/homebrew/Cellar/agda/2.6.3/libexec/ghc-9.4.4/Agd-2.6.3-d4728884/share/lib/prim/Everything.lagda
```

Should I be running `agda` on something else or with some additional flags? Thanks!

**Comment by Saransh-cpp** [comment]
Ah, thanks! Works!

**Comment by Saransh-cpp** [comment]
An updated list for reference (will create a PR shortly) - https://gist.github.com/Saransh-cpp/99476398a0bf6fa78f1fc6a26edbb9fc

**Comment by JacquesCarette** [comment]
I hope that @Saransh-cpp was using `README/Design/Fixity.agda` to set the fixities (I thought a reference to that file was here in the comments, but not so, only inside the linked PR).

**Comment by MatthewDaggitt** [comment]
Ah I'd forgotten that even existed! Maybe that belongs in the style guide rather than an Agda file? 

Anyway @Saransh-cpp, just to check does your PR follow these suggestions? :smile:

**Comment by JacquesCarette** [comment]
If you're happy to "bless" this fixity guide then yes, probably should be integrated into the style guide.

**Comment by Saransh-cpp** [comment]
Oops, just saw the comments. I was indeed following the design guide!

**Comment by JacquesCarette** [comment]
I will attempt to look at the gist seriously as soon as I can. Still buried with many other things on my plate.

**Comment by jamesmckinna** [comment]
Can the (updated) list be *sorted*, please?

**Comment by Saransh-cpp** [comment]
Should be sorted now!

**Comment by jamesmckinna** [comment]
@JacquesCarette does/did #2386 also update `README.Design.Fixity` to reflect the recent additions/adjustments?

**Comment by jamesmckinna** [comment]
The remaining 50-odd definitions without a fixity (mostly under `Function` and `Relation`...) seem much less readily to admit a classification/categorisation with 'obvious' precedences... anyone care to have a go, or should we pick a default (20, say) and go with that?

**Comment by JacquesCarette** [comment]
Yes, #2386 did update the documentation.  I was going to continue slowing picking at this issue until it's done. Yes, it is getting trickier!
